source/common/network/common_connection_filter_states.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "envoy/common/execution_context.h"
+#include "envoy/network/connection.h"
+#include "envoy/stream_info/filter_state.h"
+
+namespace Envoy {
+namespace Network {
+
+static constexpr absl::string_view kConnectionExecutionContextFilterStateName =
+    "envoy.network.connection_execution_context";
+
+// ConnectionExecutionContextFilterState is an optional connection-level filter state that goes by
+// the name kConnectionExecutionContextFilterStateName. It owns a ExecutionContext, whose
+// activate/deactivate methods will be called when a thread starts/finishes running code on behalf
+// of the corresponding connection.
+class ConnectionExecutionContextFilterState : public Envoy::StreamInfo::FilterState::Object {
+public:
+  // It is safe, although useless, to set execution_context to nullptr.
+  explicit ConnectionExecutionContextFilterState(
+      std::unique_ptr<ExecutionContext> execution_context)
+      : execution_context_(std::move(execution_context)) {}
+
+  const ExecutionContext* execution_context() const { return execution_context_.get(); }
```

Comment:
`ExecutionContext* executionContext() const { return execution_context_.get(); }`

By change this signature, you needn't two `getConnectionExecutionContextMutable/ReadOnly` methods.

Reply:
I mean you can keep only one of the two methods and needn't to do the const cast because the executeContext() could return a non-const pointer.